samples/README.md

```diff
@@ -164,27 +221,106 @@ To run this sample using [SoftHSM2](https://www.opendnssec.org/softhsm/) as the
 
 5)  Now you can run the sample:
     ```sh
-    ./pkcs11-pub-sub --endpoint <xxxx-ats.iot.xxxx.amazonaws.com> --ca_file <AmazonRootCA.pem> --cert <certificate.pem.crt> --pkcs11_lib <libsofthsm2.so> --pin <user-pin> --token_label <token-label> --key_label <key-label>
+    ./pkcs11-connect --endpoint <xxxx-ats.iot.xxxx.amazonaws.com> --ca_file <AmazonRootCA.pem> --cert <certificate.pem.crt> --pkcs11_lib <libsofthsm2.so> --pin <user-pin> --token_label <token-label> --key_label <key-label>
     ```
 
+Your Thing's [Policy](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html) must provide privileges for this sample to connect.
 
-## Raw MQTT Pub-Sub
+<details>
+<summary>(see sample policy)</summary>
+<pre>
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iot:Connect"
+      ],
+      "Resource": [
+        "arn:aws:iot:<b>region</b>:<b>account</b>:client/test-*"
+      ]
+    }
+  ]
+}
+</pre>
+</details>
 
-This sample is similar to the [Basic Pub-Sub](#basic-mqtt-pub-sub), but the connection setup is more manual.
+## Raw Connect
+
+This sample is similar to the [Basic Connect](#basic-connect), but the connection setup is more manual.
 This is a starting point for using custom
 [Configurable Endpoints](https://docs.aws.amazon.com/iot/latest/developerguide/iot-custom-endpoints-configurable.html).
 
-source: `samples/mqtt/raw_pub_sub/main.cpp`
+source: `samples/mqtt/raw_connect/main.cpp`
 
 To run the Raw MQTT Pub-Sub sample use the following command:
 
 ``` sh
-./raw-pub-sub --endpoint <endpoint> --ca_file <path to root CA>
+./raw-connect --endpoint <endpoint> --ca_file <path to root CA>
 --cert <path to the certificate> --key <path to the private key>
---topic <topic name> --user_name <user name to send on connect> --password <password to send on connect>
+--user_name <user name to send on connect> --password <password to send on connect>
 ```
 
-This will allow you to run the program. To disconnect and exit the program, enter `exit`.
+Your Thing's [Policy](https://docs.aws.amazon.com/iot/latest/developerguide/iot-policies.html) must provide privileges for this sample to connect.
+
+<details>
+<summary>(see sample policy)</summary>
+<pre>
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iot:Connect"
+      ],
+      "Resource": [
+        "arn:aws:iot:<b>region</b>:<b>account</b>:client/test-*"
+      ]
+    }
+  ]
+}
+</pre>
+</details>
+
+## x509 Credentials Provider Connect
+
+This sample is similar to the [Basic Pub-Sub](#basic-pub-sub), but the connection uses a X.509 certificate
+to prove client identity during the websocket handshake rather than during the TLS handshake.
```

Comment:
This isn't quite accurate.  The X509 certificate is used to source AWS credentials.  The websocket handshake is not change in any way, just the credential source.